ValidationRelay/ContentView.swift

```diff
@@ -10,49 +10,44 @@ import SwiftUI
 struct ContentView: View {
     @AppStorage("autoConnect") private var wantRelayConnected = false
     @AppStorage("keepAwake") private var keepAwake = true
-    
     @AppStorage("selectedRelay") private var selectedRelay = "Beeper"
     @AppStorage("customRelayURL") private var customRelayURL = ""
-    
-    @ObservedObject var relayConnectionManager: RelayConnectionManager
-    
-    init(relayConnectionManager: RelayConnectionManager) {
-        self.relayConnectionManager = relayConnectionManager
-        if wantRelayConnected {
-            relayConnectionManager.connect(getCurrentRelayURL())
-        }
-        if keepAwake {
-            UIApplication.shared.isIdleTimerDisabled = true
-        }
+
+    @StateObject private var relayConnectionManager = RelayConnectionManager()
+    @State private var customURLError: String?
+    @State private var isConnecting: Bool = false
+    @State private var connectionError: String?
+    @State private var isDimmingDisplay: Bool = false  // Separate loading state for dimming
+
+    init() {
+        // Initialization is handled by @StateObject
     }
-    
-    func getCurrentRelayURL() -> URL {
+
+    private func getCurrentRelayURL() -> URL? {
         if selectedRelay == "Custom" {
-            if let url = URL(string: customRelayURL) {
-                return url
+            guard !customRelayURL.isEmpty, let url = URL(string: customRelayURL) else {
+                customURLError = "Invalid URL"
```

Comment:
The old way of indicating an error was to just force the slider over to the default "Beeper"-- considering it is rare for users to ever enter a custom URL, this extra error handling seems a bit overcomplicated